calblueprint/meepanyar
# [CustomerData Refactor] CustomerData has Payments and MeterReadings as EntityAdapters

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
This PR changes `customerData` so that it's structurally identical to `inventoryData`. This is mostly done so that all slices in the redux store are stored similarly, and so operations done on different slices (such as adding, removing, etc...) look similar logically. The schema was also regenerated because 2 custom objects `meterReadings` and `payments` are sent from the backend.

This refactor was done now because we're about to complete functionality for payments and meter readings

Changelist
- `meterReadings` and `payments` are no longer stored inside of `customer` objects
- `meterReadings` and `payments` have their own EntityAdapters inside of `customerData`
- Schema was regenerated to fix foreign key issues
- Patches were regenerated to include new customer objects `meterReadings` and `payments`
- New selectors were created for `meterReadings` and `payments`
- Files `Home.tsx`, `CustomerMain.tsx`, `CustomerProfile.tsx` and `CustomerUtils.tsx` were refactored so that they used the selectors created for `meterReadings` and `payments` 

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## How to review
1. Use the correct backend branch `julian/customer-data-refactor` PR'd [here](https://github.com/calblueprint/meepanyar-node/pull/22)
2. Open the app and clear site data to reset redux
3. Log in and make sure there are no errors on the Home, CustomerMain, CustomerProfile, or Records screens
4. Create a customer and make sure that they are still properly made on the frontend and backend
5. Edit a customer and make sure they're still properly editable
6. Go to the customer's records and make sure that meter readings and payments are populated

[//]: # 'The order in which to review files and what to expect when testing locally'

## Relevant Links

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## Next steps
- Refactor sites
- Functionality for payments and meter readings

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @julianrkung

[//]: # 'This tags Julian as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
